src/cascadia/TerminalApp/AppLogic.h

```diff
@@ -56,124 +46,47 @@ namespace winrt::TerminalApp::implementation
-        void DismissDialog();
 
         Windows::Foundation::Collections::IMapView<Microsoft::Terminal::Control::KeyChord, Microsoft::Terminal::Settings::Model::Command> GlobalHotkeys();
 
         Microsoft::Terminal::Settings::Model::Theme Theme();
 
-        // -------------------------------- WinRT Events ---------------------------------
-        // PropertyChanged is surprisingly not a typed event, so we'll define that one manually.
-        // Usually we'd just do
-        //    WINRT_CALLBACK(PropertyChanged, Windows::UI::Xaml::Data::PropertyChangedEventHandler);
-        //
-        // But what we're doing here is exposing the Page's PropertyChanged _as
-        // our own event_. It's a FORWARDED_CALLBACK, essentially.
-        winrt::event_token PropertyChanged(Windows::UI::Xaml::Data::PropertyChangedEventHandler const& handler) { return _root->PropertyChanged(handler); }
-        void PropertyChanged(winrt::event_token const& token) { _root->PropertyChanged(token); }
+        TerminalApp::TerminalWindow CreateNewWindow();
 
-        TYPED_EVENT(RequestedThemeChanged, winrt::Windows::Foundation::IInspectable, winrt::Microsoft::Terminal::Settings::Model::Theme);
-        TYPED_EVENT(SettingsChanged, winrt::Windows::Foundation::IInspectable, winrt::Windows::Foundation::IInspectable);
-        TYPED_EVENT(SystemMenuChangeRequested, winrt::Windows::Foundation::IInspectable, winrt::TerminalApp::SystemMenuChangeArgs);
+        TYPED_EVENT(SettingsChanged, winrt::Windows::Foundation::IInspectable, winrt::TerminalApp::SettingsLoadEventArgs);
 
     private:
         bool _isUwp{ false };
         bool _isElevated{ false };
 
-        // If you add controls here, but forget to null them either here or in
-        // the ctor, you're going to have a bad time. It'll mysteriously fail to
-        // activate the AppLogic.
-        // ALSO: If you add any UIElements as roots here, make sure they're
-        // updated in _ApplyTheme. The root currently is _root.
-        winrt::com_ptr<TerminalPage> _root{ nullptr };
         Microsoft::Terminal::Settings::Model::CascadiaSettings _settings{ nullptr };
 
         winrt::hstring _settingsLoadExceptionText;
         HRESULT _settingsLoadedResult = S_OK;
         bool _loadedInitialSettings = false;
 
-        uint64_t _numOpenWindows{ 0 };
-
-        std::shared_mutex _dialogLock;
-        winrt::Windows::UI::Xaml::Controls::ContentDialog _dialog;
-
-        ::TerminalApp::AppCommandlineArgs _appArgs;
+        bool _hasSettingsStartupActions{ false };
         ::TerminalApp::AppCommandlineArgs _settingsAppArgs;
 
         std::shared_ptr<ThrottledFuncTrailing<>> _reloadSettings;
         til::throttled_func_trailing<> _reloadState;
 
+        winrt::Windows::Foundation::Collections::IVector<Microsoft::Terminal::Settings::Model::SettingsLoadWarnings> _warnings{ winrt::multi_threaded_vector<Microsoft::Terminal::Settings::Model::SettingsLoadWarnings>() };

```

Comment:
You should avoid using the WinRT collection types. `IVector` is basically like using a linked list, performance wise. Instead, as far as I can tell, you can just wrap it into a `winrt::multi_threaded_vector` whenever its actually needed. You can avoid making a copy of `_warnings` by using `std::move`, since `_warnings` is only read exactly once anyways, after a settings reload.